File: Android/common/unpack.py
from optparse import OptionParser 
import zipfile
import os 
import sys 

src=''
dest='./'

def main():
    parser = OptionParser()
    parser.add_option('-s' ,'--src' , dest="src" , help='apk file path')
    parser.add_option('-o' , '--output',dest="output", help='dex file output path')

    (options ,args) = parser.parse_args()

    if options.src:
        src = options.src
    else: 
        exit() 

    if options.output:
        dest=options.output
        if not dest.endswith('/'):
            dest+='/'

    os.system('unzip -o '+ src+' -d '+ dest+'out')
    paths=os.listdir(dest+'out')
    for path in paths:
        if path.endswith('.dex'):
            dexdir=dest+path
            if not os.path.exists(dexdir):
                os.mkdir(dest+path)
            os.system('cp '+dest+'out/'+path+" "+dexdir )
            os.system('java -jar common/ShakaApktool.jar bs '+dest+path+'/'+path+" -o "+dexdir+"/out")


if __name__ == '__main__':
    main()

File: Android/common/unpack.js
var src=''
var dest='./'
var child=require('child_process')
var path = require('path')
var fs = require('fs')

function execSystem(){
    child.execSync('unzip -o '+src+' -d '+ dest+'out')
    paths=fs.readdirSync(dest+'out')
    paths.forEach(function(ele,index){
        if(ele.endsWith('.dex')){
            dexdir=dest+ele 
            console.log(dexdir)
            if(!fs.existsSync(dexdir)){
                child.execSync('mkdir '+ dexdir)
            }

            child.execSync('cp '+dest+'out/'+ele+' '+dexdir)
            // TODO the process  end before child finish its work
            child.execSync( '  java -jar common/ShakaApktool.jar bs '+dest+path+'/'+path+" -o "+dexdir+"/out",
            function(err,stdout,stderr){
                console.log('sssss' )
            })
            console.log('unpack ok')    
         }
    })
}

function main(){
    if(process.argv.length < 4){
        console.log('ggggg')
        return 
    } 

    src = process.argv[2]
    dest = process.argv[3]
    if (!dest.endsWith('/')){
        dest+='/'
    }

    execSystem()
}





main()
